pkg/api/handlers/projectinvitation.go

```diff
@@ -0,0 +1,332 @@
+			Status:   string(types.ProjectInvitationStatusPending),
+			ThreadID: thread.Name,
+		},
+	}
+
+	if err := req.Create(&invitation); err != nil {
+		return err
+	}
+
+	return req.WriteCreated(types.ProjectInvitationManifest{
+		Code:    code,
+		Project: &project,
+		Status:  types.ProjectInvitationStatusPending,
+		Created: invitation.CreationTimestamp.Format(time.RFC3339),
+	})
+}
+
+func (h *ProjectInvitationHandler) ListInvitationsForProject(req api.Context) error {
+	thread, err := getProjectThread(req)
+	if err != nil {
+		return err
+	}
+
+	if !thread.Spec.Project {
+		return types.NewErrBadRequest("only projects can have invitations")
+	}
+
+	if thread.Spec.UserID != req.User.GetUID() {
+		return types.NewErrForbidden("only the project creator can list invitations")
+	}
+
+	var project types.Project
+	if thread.Spec.ParentThreadName != "" {
+		var parentThread *v1.Thread
+		if err := req.Get(parentThread, thread.Spec.ParentThreadName); err != nil {
+			return err
+		}
+		project = convertProject(thread, parentThread)
+	} else {
+		project = convertProject(thread, nil)
+	}
+
+	var invitations v1.ProjectInvitationList
+	if err := req.List(&invitations, kclient.MatchingFields{
+		"spec.threadID": thread.Name,
+	}); err != nil {
+		return err
+	}
+
+	// Convert to list of manifests
+	manifests := make([]types.ProjectInvitationManifest, len(invitations.Items))
+	for i, invitation := range invitations.Items {
+		manifests[i] = types.ProjectInvitationManifest{
+			Code:    invitation.Name,
+			Project: &project,
+			Status:  types.ProjectInvitationStatus(invitation.Spec.Status),
+			Created: invitation.CreationTimestamp.Format(time.RFC3339),
+		}
+	}
+
+	return req.Write(manifests)
+}
+
+func (h *ProjectInvitationHandler) DeleteInvitationForProject(req api.Context) error {
+	var (
+		code = req.PathValue("code")
+	)
+
+	thread, err := getProjectThread(req)
+	if err != nil {
+		return err
+	}
+
+	if !thread.Spec.Project {
+		return types.NewErrBadRequest("only projects can have invitations")
+	}
+
+	if thread.Spec.UserID != req.User.GetUID() {
+		return types.NewErrForbidden("only the project creator can delete invitations")
+	}
+
+	var invitation v1.ProjectInvitation
+	if err := req.Get(&invitation, code); err != nil {
+		if apierrors.IsNotFound(err) {
+			return types.NewErrNotFound("invitation not found")
+		}
+		return err
+	}
+
+	// Verify the invitation belongs to this project
+	if invitation.Spec.ThreadID != thread.Name {
+		return types.NewErrBadRequest("invitation does not belong to this project")
+	}
+
+	if err := req.Delete(&invitation); err != nil {
+		return err
+	}
+
+	return req.Write(nil)
+}
+
+func (h *ProjectInvitationHandler) GetInvitation(req api.Context) error {
+	var (
+		code       = req.PathValue("code")
+		invitation v1.ProjectInvitation
+	)
+
+	if err := req.Get(&invitation, code); err != nil {
+		if apierrors.IsNotFound(err) {
+			return types.NewErrNotFound("invitation not found")
+		}
```

Comment:
If you just return the error, then the framework will detect the `NotFoundError` and do the same thing here.

There are a few places where you do this.

Reply:
fixed